### ISSUE TITLE
processing BHL PDF

### ISSUE THREAD

**Issue by myrmoteras** [plazi/GoldenGATE-Imagine#28]
Let's have a discussion on how to process BHL pdfs like this one

* needs also an effort how to decode the BHL versions of it
[opinionDeclarationICZN.462.pdf](https://github.com/plazi/Plazi-Communications/files/10087541/opinionDeclarationICZN.462.pdf)
https://tb.plazi.org/GgServer/summary/FFCBF110FFEEFFDDFFE8FFCAFFC4385C 

https://www.biodiversitylibrary.org/page/34656957

http://biostor.org/reference/65852 (the source)

Since this pdf did not process in GGI 

[GgImagine.20221125-0852.out.zip](https://github.com/plazi/GoldenGATE-Imagine/files/10089659/GgImagine.20221125-0852.out.zip)

I then run it through ABBYY which then processed (see above)
[opinionDeclarationICZN.462.abbyy.pdf](https://github.com/plazi/GoldenGATE-Imagine/files/10089672/opinionDeclarationICZN.462.abbyy.pdf)

**Comment by gsautter** [comment]
This error message should actually tell you what went wrong on the first attempt: "Too few words per page (0 on 12 pages, less than 25)", which only ever comes up if you decode a PDF as born-digital ... in that case, words covered up with other objects like figures or filled graphics are filtered out, and the page images obviously obstruct the view on the OCR words hidden underneath them ... if you open the first one as scanned with embedded OCR, it works just like supposed to.

**Comment by myrmoteras** [comment]
I am pretty sure, I used this option. For sure not born digital. This happens now the second time. Did you actually decode the file and use this option. 
I will check later again
D

**Comment by myrmoteras** [comment]
so what is the option to choose.
![image](https://user-images.githubusercontent.com/4609956/204057279-aaedf482-f02c-49da-aa6b-c7654e816265.png)

**Comment by gsautter** [comment]
> I am pretty sure, I used this option. For sure not born digital. This happens now the second time. Did you actually decode the file and use this option. I will check later again D

Yes, the first PDF you linked (`opinionDeclarationICZN.462.pdf`) did actually open as "scanned with embedded OCR" ... if you get "Too few words per page", it's always an indicator of trying to decode a scanned PDF as born-digital.

**Comment by gsautter** [comment]
> so what is the option to choose. ![image](https://user-images.githubusercontent.com/4609956/204057279-aaedf482-f02c-49da-aa6b-c7654e816265.png)

"PDF Documents (scanned)" should do the trick.

**Comment by gsautter** [comment]
There is an issue with black/white inversion, though ... checking it out right now.

**Comment by myrmoteras** [comment]
ok, when to use 
* PDF Documents (scanned, text digitized) - the obvious version for every PDF which has been scanned and ocred.
* PDF Documents (scanned, text vectorized)
* PDF Documents (scanned) - the obvious version for every PDF where the no OCR has been applied, just an iimage is availble.
*
![PXL_20221125_213507509](https://user-images.githubusercontent.com/4609956/204057559-b10cdcea-f602-4b1f-9cdf-71957672f03d.jpg)

**Comment by gsautter** [comment]
> ok, when to use
> 
>     * PDF Documents (scanned, text digitized) - the obvious version for every PDF which has been scanned and ocred.
> 
>     * PDF Documents (scanned, text vectorized)
> 
>     * PDF Documents (scanned) - the obvious version for every PDF where the no OCR has been applied, just an iimage is availble.

Think I detailed this out in an earlier ticket ... "PDF Documents (scanned)" is what you want to use in the very most cases, embedded OCR or not ...
The other two you need only **if the text is not in the page images anymore** at all (only seen in a few of the Brazilian PDFs earlier this year so far), "PDF Documents (scanned, text vectorized)" if the text is represented in DjVu style encoding (still looking like the original text), and "PDF Documents (scanned, text digitized)" if the text is regular PDF text rendering with normal fonts like in a born-digital PDF.

**Comment by myrmoteras** [comment]
now I am running PDF Documents (scanned)

this is producing something completely garbeled.
![image](https://user-images.githubusercontent.com/4609956/204058006-01597fb2-34df-4a7c-957f-237bd42b4580.png)
[opinionDeclarationICZN.462.pdf.imdir.zip](https://github.com/plazi/GoldenGATE-Imagine/files/10095166/opinionDeclarationICZN.462.pdf.imdir.zip)
[GgImagine.20221125-2232.err.zip](https://github.com/plazi/GoldenGATE-Imagine/files/10095168/GgImagine.20221125-2232.err.zip)

**Comment by gsautter** [comment]
Weird ... but I think that's actually a downstream effect of an inverted page image (see above) in combination with embedded OCR adjustment - with the latter deactivated, I got a pretty good result.

But anyway, I take it the ABBYY version decoded without incident?

**Comment by myrmoteras** [comment]
nd "PDF Documents (scanned, text digitized)" if the text is regular PDF text rendering with normal fonts like in a born-digital PDF.

This is the one that works, when I am running a scanned PDF that I run through ABBYY, like the one above - I include the example here again 
[opinionDeclarationICZN.462.abbyy.pdf](https://github.com/plazi/GoldenGATE-Imagine/files/10095172/opinionDeclarationICZN.462.abbyy.pdf)

however, all the BHL PDFs are also OCRed, and thus should be processed using the same setting ?  PDF Documents (scanned, text digitized)"

May be this is another one that has the wrong image size setting?

**Comment by myrmoteras** [comment]
The ABBBY decoded file, though with some OCR errors that appear in GGI that do not exist in the orignal OCR-ed PDF, has been processed and is in TB. https://tb.plazi.org/GgServer/summary/FFCBF110FFEEFFDDFFE8FFCAFFC4385C

**Comment by gsautter** [comment]
> Think I detailed this out in an earlier ticket ... "PDF Documents (scanned)" is what you want to use in the very most cases, embedded OCR or not ...
> The other two you need only **if the text is not in the page images anymore** at all (only seen in a few of the Brazilian PDFs earlier this year so far), "PDF Documents (scanned, text vectorized)" if the text is represented in DjVu style encoding (still looking like the original text), and "PDF Documents (scanned, text digitized)" if the text is regular PDF text rendering with normal fonts like in a born-digital PDF.

See above ...

**Comment by myrmoteras** [comment]
So, my conclusion is, that I can't process this  (and other ) BHL born PDF, since nothing works.

It means, we need to 
* add an ABBYY step
*  find out how to solve the BHL PDF issues
*  understand and talk again to bHL to be able to scale up processing
*  find a way to transform the PDF, e.g. change the setting of the image size

**Comment by gsautter** [comment]
> * find out how to solve the BHL PDF issues

That's what I'm working on right now ... seems to be something with the way mask images are handled (scan decomposed into low-resolution color background and high resolution monochrome text mask).

**Comment by gsautter** [comment]
Finally figured it out ... BHL PDFs have a very peculiar way of encoding the text layer of a page image, which the decoder didn't handle the completely correct way ... the original one decodes nicely now, without the ABBYY detour.

**Comment by myrmoteras** [comment]
Tx. Will check with some other BHL pdfs and let you know

New species of the day<https://tb.plazi.org/GgServer/static/newToday.html>
________________________________

**Comment by myrmoteras** [comment]
please let me know, when the update is available in GGI.

**Comment by gsautter** [comment]
Will do ... just for info: is there any upcoming conversation with BHL we need this deployed for?

**Comment by myrmoteras** [comment]
No. No BHL I conversation planned. Don't we have updates for GGi in regular intervals?

New species of the day<https://tb.plazi.org/GgServer/static/newToday.html>
________________________________

**Comment by gsautter** [comment]
Well, I put out the updates as needed ... given the last weeks were mostly data processing (MSW, CoL, etc.) and meetings, there wasn't all too much to update ... will put this one out soon as I'm done with the Type3 font implementation (https://github.com/plazi/ggi/issues/284).

**Comment by myrmoteras** [comment]
sorry to have brought this up again in https://github.com/plazi/GoldenGATE-Imagine/issues/31

But, yes, there is an attempt to talk to BHL and find out their latest status on OCR-ing. 

But may be we can fix the displaying to the BHL pages.

**Comment by gsautter** [comment]
Might this be fixed with https://github.com/plazi/GoldenGATE-Imagine/issues/31 ? Sure looks awfully similar ...
